tools/topology/sof/pipe-mux-capture.m4

```diff
@@ -0,0 +1,80 @@
+# Passthrough Capture Pipeline with MUX and PCM
+#
+# Pipeline Endpoints for connection are :-
+#
+#  host PCM_C <-+- Mux 0 <-- B0 <-- source DAI0
+#               |
+#  Others    <--+
+
+# Include topology builder
+include(`utils.m4')
+include(`buffer.m4')
+include(`pcm.m4')
+include(`dai.m4')
+include(`mux.m4')
+include(`enumcontrol.m4')
+include(`pipeline.m4')
+
+#
+# Controls
+#
+# Volume Mixer control with max value of 32
+# TODO pass in enum options
+C_CONTROLENUM_TEXT(OUTPUTS, LIST(`		', STR(N_PCMC(PCM_ID)), STR(Other)))
+C_CONTROLENUM(Capture Mux, PIPELINE_ID,
+	CONTROLENUM_OPS(enum, 257 binds the mixer control to enum get/put handlers, 257, 257),
+	LIST(`	', KCONTROL_ENUM_CHANNEL(FL, 0, 0), KCONTROL_ENUM_CHANNEL(FR, 0, 1)),
+	OUTPUTS)
+
+#
+# Components and Buffers
+#
+
+# Host "Passthrough Capture" PCM
+# with 0 sink and 2 source periods
+W_PCM_CAPTURE(PCM_ID, Mux Capture, 0, 2, 2)
+
+# "Mux" has 2 source and 2 sink periods
+W_MUX(0, PIPELINE_FORMAT, 2, 2, SOF_COMP_MUX_GENERIC, LIST(`		', "Capture Mux PIPELINE_ID"), PIPELINE_ID)
+
+# Capture Buffers
+W_BUFFER(0, COMP_BUFFER_SIZE(2,
+	COMP_SAMPLE_SIZE(PIPELINE_FORMAT), PIPELINE_CHANNELS, SCHEDULE_FRAMES),
+	PLATFORM_HOST_MEM_CAP)
+# Capture Buffers
+W_BUFFER(1, COMP_BUFFER_SIZE(2,
+	COMP_SAMPLE_SIZE(PIPELINE_FORMAT), PIPELINE_CHANNELS, SCHEDULE_FRAMES),
+	PLATFORM_HOST_MEM_CAP)
+
+W_BUFFER(1, COMP_BUFFER_SIZE(2,
+	COMP_SAMPLE_SIZE(PIPELINE_FORMAT), PIPELINE_CHANNELS, SCHEDULE_FRAMES),
+	PLATFORM_HOST_MEM_CAP)
```

Comment:
is there a copy/paste here, the W_BUFFER(1 seems duplicated.

Reply:
fixed